hie-plugin-api/Haskell/Ide/Engine/ModuleCache.hs

```diff
@@ -182,8 +182,11 @@ loadCradle iniDynFlags (NewCradle fp) def action = do
         liftIO (GHC.newHscEnv iniDynFlags) >>= GHC.setSession
         liftIO $ setCurrentDirectory (BIOS.cradleRootDir cradle)
         init_res <- gcatches (Right <$> init_session)
-                [ErrorHandler (\(ex :: GHC.GhcException)
-                  -> return $ Left (GHC.showGhcException ex ""))]
+                [ ErrorHandler (\(ex :: AsyncException)
+                   -> liftIO (throwIO ex))
+                , ErrorHandler (\(ex :: SomeException)
+                   -> return $ Left (show ex))
+                ]
```

Comment:
Eh, still opposed to handling Async Exceptions. I dont think we should try to catch all exceptions. In some cases, we just cant continue and crashing is the most sensible option.

Reply:
Ok, so i will catch only `SourceError` (and `IOException`?)

Reply:
I think, I would prefer that. What do @mpickering, @alanz, etc.. think?

Reply:
Agree, rather build up a list of the ones we know it is safe to deal with. Which we had (maybe still do?) on the handler for the call to GHC when compiling.

Reply:
Thanks for the tip @alanz: i found it and it seems to be just what we needed here: https://github.com/mpickering/haskell-ide-engine/blob/hie-bios/hie-plugin-api/Haskell/Ide/Engine/Ghc.hs#L203